core/dbt/graph/graph.py

```diff
@@ -59,9 +59,36 @@ def select_childrens_parents(self, selected: Set[UniqueId]) -> Set[UniqueId]:
     def select_children(
         self, selected: Set[UniqueId], max_depth: Optional[int] = None
     ) -> Set[UniqueId]:
-        descendants: Set[UniqueId] = set()
-        for node in selected:
-            descendants.update(self.descendants(node, max_depth))
+        """Return all descendants of the selected nodes, up to max_depth levels deep"""
+        related_nodes = selected
+        current_layer = selected
+        current_depth = 0
+
+        # this process steps through the graph, layer by layer, until max_depth or no new nodes are found.
+        # going 1 step at a time allows for pruning of the graph since many selected nodes may overlap in descendants
+
+        if max_depth is None:
+            max_depth = 999999999
```

Comment:
this is temp until I get more review/testing, unsure exactly how to get flake8 to be satisfied to compare to `Optional[int]` in the loop